aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -314,19 +314,25 @@ void prepare_matrices_for_broadcasting(
           MPSGraphTensor* productTimesAlphaTensor = [mpsGraph multiplicationWithPrimaryTensor:productTensor
                                                                               secondaryTensor:alphaTensor
                                                                                          name:@"MM/alpha*(mat1@mat2)"];
-          MPSGraphTensor* biasTimesBetaTensor = [mpsGraph multiplicationWithPrimaryTensor:biasTensor
-                                                                          secondaryTensor:betaTensor
-                                                                                     name:@"MM/beta*input"];
+          MPSGraphTensor* biasTimesBetaTensor = biasTensor;
+          if (beta.toDouble() != 0.0) {
```

Comment:
I think you can hoist this condition out and create a boolean out of this to add the checks here in if-condition.